overlord/hookstate/ctlcmd/get_test.go

```diff
@@ -205,115 +237,112 @@ func (s *getAttrSuite) SetUpTest(c *C) {
 	ch.AddTask(slotHookTask)
 }
 
-func (s *getAttrSuite) TestGetPlugAttributesInPlugHook(c *C) {
-	stdout, stderr, err := ctlcmd.Run(s.mockPlugHookContext, []string{"get", ":aplug", "aattr"})
-	c.Check(err, IsNil)
-	c.Check(string(stdout), Equals, "foo\n")
-	c.Check(string(stderr), Equals, "")
-
-	stdout, stderr, err = ctlcmd.Run(s.mockPlugHookContext, []string{"get", "-d", ":aplug", "baz"})
-	c.Check(err, IsNil)
-	c.Check(string(stdout), Equals, "{\n\t\"baz\": [\n\t\t\"a\",\n\t\t\"b\"\n\t]\n}\n")
-	c.Check(string(stderr), Equals, "")
-
-	// The --plug parameter doesn't do anything if used on plug side
-	stdout, stderr, err = ctlcmd.Run(s.mockPlugHookContext, []string{"get", "--plug", ":aplug", "aattr"})
-	c.Check(err, IsNil)
-	c.Check(string(stdout), Equals, "foo\n")
-	c.Check(string(stderr), Equals, "")
-}
-
-func (s *getAttrSuite) TestGetSlotAttributesInSlotHook(c *C) {
-	stdout, stderr, err := ctlcmd.Run(s.mockSlotHookContext, []string{"get", ":bslot", "battr"})
-	c.Check(err, IsNil)
-	c.Check(string(stdout), Equals, "bar\n")
-	c.Check(string(stderr), Equals, "")
-
-	// The --slot parameter doesn't do anything if used on slot side
-	stdout, stderr, err = ctlcmd.Run(s.mockSlotHookContext, []string{"get", "--slot", ":bslot", "battr"})
-	c.Check(err, IsNil)
-	c.Check(string(stdout), Equals, "bar\n")
-	c.Check(string(stderr), Equals, "")
-}
-
-func (s *getAttrSuite) TestGetSlotAttributeInPlugHook(c *C) {
-	stdout, stderr, err := ctlcmd.Run(s.mockPlugHookContext, []string{"get", "--slot", ":aplug", "battr"})
-	c.Check(err, IsNil)
-	c.Check(string(stdout), Equals, "bar\n")
-	c.Check(string(stderr), Equals, "")
-}
-
-func (s *getAttrSuite) TestGetPlugAttributeInSlotHook(c *C) {
-	stdout, stderr, err := ctlcmd.Run(s.mockSlotHookContext, []string{"get", "--plug", ":bslot", "aattr"})
-	c.Check(err, IsNil)
-	c.Check(string(stdout), Equals, "foo\n")
-	c.Check(string(stderr), Equals, "")
-}
-
-func (s *getAttrSuite) TestUnknownPlugAttribute(c *C) {
-	stdout, stderr, err := ctlcmd.Run(s.mockPlugHookContext, []string{"get", ":aplug", "x"})
-	c.Check(err, NotNil)
-	c.Check(err.Error(), Equals, `unknown attribute "x"`)
-	c.Check(string(stdout), Equals, "")
-	c.Check(string(stderr), Equals, "")
-}
-
-func (s *getAttrSuite) TestUnknownSlotAttribute(c *C) {
-	stdout, stderr, err := ctlcmd.Run(s.mockSlotHookContext, []string{"get", ":bslot", "x"})
-	c.Check(err, NotNil)
-	c.Check(err.Error(), Equals, `unknown attribute "x"`)
-	c.Check(string(stdout), Equals, "")
-	c.Check(string(stderr), Equals, "")
-}
-
-func (s *getAttrSuite) TestUsingPlugNameInSlotHookFails(c *C) {
-	stdout, stderr, err := ctlcmd.Run(s.mockSlotHookContext, []string{"get", ":aplug", "x"})
-	c.Check(err, NotNil)
-	c.Check(err.Error(), Equals, `unknown plug or slot "aplug"`)
-	c.Check(string(stdout), Equals, "")
-	c.Check(string(stderr), Equals, "")
+var getPlugAttributesTests = []struct {
+	args, stdout, error string
+}{
+	{
+		args:   "get :aplug aattr",
+		stdout: "foo\n",
+	},
+	{
```

Comment:
For that sort of table, this format saves both vertical and horizontal spacing, rendering the code more readable:
```
...
}{{
    args: ...
    stdout: ...
}, {
    args: ...
    stdout: ...
}, {
...
```